### DISCUSSION TITLE
is it able to use thirdpart bars such as waybar?

### DISCUSSION THREAD

**Discussion by superiums** [qtile/qtile#4507] [General]
from now on, i realize my most crash of qtile was caused my Widgets. 
and also, the widgets has some function leak.

so i wonder wheath it's possible to use waybar ?

**Comment by elParaguayo** [comment]
Should be possible. Have you tried it?

**Reply by superiums** [reply]
i've tried it. 
seems there's some proble with comunication of these two.
the `systray`, `wlr/workspaces` and `wlr/taskbar` was not able to communicate with qtile.

other widgets seems ok.

**Comment by elParaguayo** [comment]
Also, could you let me know which widgets have a "function leak" so we can take a look at fixing this?

**Reply by superiums** [reply]
i'm on wayland, so main face some problem in wayland.

for example, systray not avaluable, so i use StatusNotifier, but it's not interactive (maybe i'm not familiar with it).

- the most headache was some widget will crash all, and no change to switch to any other tty or quit system. in other desktop, if crash, we are able to switch to other tty. as i reported , after tens tries, finally i found it is: volume, and chord.

- the second headache was my wireplumber/xwayland sometimes takes over the cpu. but this not happen in sway/river. after reduce lots of widgets, seems this not happen so often.

**Reply by elParaguayo** [reply]
Thanks for taking the time to reply.

> for example, systray not avaluable, so i use StatusNotifier, but it's not interactive (maybe i'm not familiar with it).

Yes, as discussed separately, `Systray` is X11 only. For `StatusNotifier`, you're correct that the stock widget is non-interactive. However, the docs say this: "Context menus are not currently supported by the official widget. However, a modded version of the widget which provides basic menu support is available from elParaguayo's [qtile-extras](https://github.com/elParaguayo/qtile-extras) repo."

> * the most headache was some widget will crash all, and no change to switch to any other tty or quit system. in other desktop, if crash, we are able to switch to other tty. as i reported , after tens tries, finally i found it is: volume, and chord.

This one is odd. I can't see any good reasons for these to crash. Will need to investigate more.

> * the second headache was my wireplumber/xwayland sometimes takes over the cpu. but this not happen in sway/river. after reduce lots of widgets, seems this not happen so often.

Yeah. Also frustrating. Pinging @m-col and @jwijenbergh so they're aware.

**Reply by m-col** [reply]
Thanks. I've recently been seeing high CPU usage due to XWayland. That it doesn't happen in other compositors is good to know, thanks. Will investigate.

**Reply by superiums** [reply]
Thanks for all yours effects, it will become more and more great !
